pkg/skaffold/verify/k8sjob/verify.go

```diff
@@ -0,0 +1,461 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sjob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/fatih/semgroup"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/trace"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubectl/pkg/scheme"
+
+	component "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/component/kubernetes"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/kubectl"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/label"
+	eventV2 "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/event/v2"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/instrumentation"
+	k8sjoblogger "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/k8sjob/logger"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/k8sjob/tracker"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/kubernetes"
+	kubernetesclient "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/kubernetes/client"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/loader"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/log"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/status"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/util"
+)
+
+// Verifier verifies deployments using kubernetes libs/CLI.
+type Verifier struct {
+	configName string
+
+	cfg              []*latest.KubernetesClusterVerifyTestCase
+	tracker          *tracker.JobTracker
+	imageLoader      loader.ImageLoader
+	logger           *k8sjoblogger.Logger
+	statusMonitor    status.Monitor
+	originalImages   []graph.Artifact // the set of images parsed from the Verifier's manifest set
+	localImages      []graph.Artifact // the set of images marked as "local" by the Runner
+	kubectl          kubectl.CLI
+	labeller         *label.DefaultLabeller
+	defaultNamespace string
+}
+
+// NewVerifier returns a new Verifier for a VerifyConfig filled
+// with the needed configuration for `kubectl apply`
+func NewVerifier(ctx context.Context, cfg kubectl.Config, labeller *label.DefaultLabeller, v *latest.KubernetesClusterVerifier, artifacts []*latest.Artifact) (*Verifier, error) {
+	defaultNamespace := ""
+	b, err := (&util.Commander{}).RunCmdOut(context.Background(), exec.Command("kubectl", "config", "view", "--minify", "-o", "jsonpath='{..namespace}'"))
+	if err == nil {
+		defaultNamespace = strings.Trim(string(b), "'")
+		if defaultNamespace == "default" {
+			defaultNamespace = ""
+		}
```

Comment:
According to comment in line 95, IIUC, I think we don't need this?

Reply:
Currently code has changed, not sure what comment you are referring to.  While a bit convoluted, currently this is neccessary IIUC as this value when used when creating the kubectl CLI must have "" and not "default"
```
	kubectl := kubectl.NewCLI(cfg, latest.KubectlFlags{}, defaultNamespace)
```
and then in the `Verify` code it must be "default" and not ""

Reply:
You're right, didn't notice. the use in `kubectl.NewCLI`, my bad.